pkg/proxy/iptables/proxier.go

```diff
@@ -764,7 +806,7 @@ func getLocalIPs(endpointsMap proxyEndpointsMap) map[types.NamespacedName]sets.S
 				if localIPs[nsn] == nil {
 					localIPs[nsn] = sets.NewString()
 				}
-				ip := strings.Split(ep.endpoint, ":")[0] // just the IP part
+				ip := ep.endpoint[0:strings.Index(ep.endpoint, ":")] //just the IP part
```

Comment:
Should we guard against strings without `:`?  We could also pre-split this, right?